src/WebSarcoClient.ts

```diff
@@ -61,18 +69,65 @@ export class WebSarcoClient {
     // TODO: Allow client to choose when to start/stop libp2p node
     await this.startLibp2pNode();
 
+    let bundlrProvider: ethers.providers.Web3Provider;
+
+    if (params.bundlrPublicKey) {
+      // Custom provider for bundlr that uses a Sarcophagus DAO server-provided public key to sign transactions to sponsor
+      // upload costs for embalmers.
+      const pubKey = Buffer.from(params.bundlrPublicKey, 'hex');
+      bundlrProvider = {
+        getPublicKey: async () => {
+          return pubKey;
+        },
+        getSigner: () => {
+          return {
+            publicKey: pubKey,
+            getAddress: () => pubKey,
+            _signTypedData: async (
+              _domain: never,
+              _types: never,
+              message: { address: string; 'Transaction hash': Uint8Array }
+            ) => {
+              let messageData = Buffer.from(message['Transaction hash']).toString('hex');
+              // const res = await fetch('http://localhost:4000/bundlr/signData', {
+              const res = await fetch('https://api.encryptafile.com/bundlr/signData', {
```

Comment:
Can this base URL `https://api.encryptafile.com/` be moved out into the network config? This domain will most likely change to one controlled by the DAO. Looks like it's also used in `src/Archaeologist.ts`

Reply:
Does this endpoint need to also include a `chainID` as a param, if we want the DAO signer to support multiple chains (testnet/mainnet)? Or is that somehow included in the `messageData`?